drivers/hmis/app/graphql/types/hmis_schema/client.rb

```diff
@@ -185,6 +192,18 @@ def enrollments(**args)
       resolve_enrollments(scope, **args, dangerous_skip_permission_check: true)
     end
 
+    def active_enrollment(project_id:, open_on_date:)
+      # TODO: data loader. not sure how to make it most efficient. Each client probably doesn't have THAT many enrollments,
+      # but this is fetched for a page of clients. The project_id (and date!) should be "constant with respect to the resolver" so maybe
+      # we can use it in the scope like below? doubt that will work, try it
+      # enrollment = load_ar_association(object, :enrollments, scope: Hmis::Hud::Enrollment.with_project(project_id).open_on_date(open_on_date).order(entry_date: :desc, date_created: :asc))&.last
+
+      object.enrollments.
+        open_on_date(open_on_date).
+        with_project(project_id).
+        order(entry_date: :desc, date_created: :asc).last
```

Comment:
Thanks for the suggestion!
I think I like your proposal, except it will miss WIP Enrollments due to the `e.project_id` comparison. Maybe could be resolved with a preload on the wip table.

I think the project + date really are constant, at least they are in the way we are using them.. I updated and added a warning about that https://github.com/greenriver/hmis-warehouse/pull/4087/commits/e5e42e926e0bd47e10fef12419c2f4dc87eda7b8 which feels ok to me - what do you think?

Reply:
Thanks Gig, I think e5e42e9 is a reasonable solution to get this out the door.

While this case isn't likely to be a problem, I'm a uncomfortable with long-term use of resolvers that return incorrect results under some circumstances. Maybe we could plan to revisit this at some point? I'm thinking we could add an issue in the ice-box to track it or use todo-or-die

Reply:
definitely. [PT story](https://www.pivotaltracker.com/story/show/187200498)